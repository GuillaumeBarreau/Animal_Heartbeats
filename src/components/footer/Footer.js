import React from 'react';
import styled from 'styled-components';
import { connect } from 'react-redux';

const FooterContent = styled.footer`
	padding: 2rem;
	margin-top: 2.5rem;
	background: ${props => props.theme.secondary};
	color: ${props => props.theme.text};
`;

const Link = styled.a`
	color: red;
	font-weight: 600;
	text-decoration: none;
	:focus {
		color: rebeccapurple;
	}
`;

const Footer = ({ theme }) => (
	<FooterContent theme={theme}>
		<p>
			Made with love &hearts; &hearts; &hearts; - Repository{' '}
			<Link target="_blank" href="https://github.com/GuillaumeBarreau/Animal_Heartbeats">
				GitHub
			</Link>
		</p>
	</FooterContent>
);

const mapStateToProps = state => {
	return {
		theme: state.theme,
	};
};

export default connect(mapStateToProps)(Footer);
