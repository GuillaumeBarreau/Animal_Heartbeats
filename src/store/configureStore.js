import { createStore, applyMiddleware } from 'redux';
import reducers from './reducers/combineReducers';

function logger({ getState }) {
	return next => action => {
		console.log('will dispatch', action);
		return next(action);
	};
}

export const store = createStore(reducers, applyMiddleware(logger));
 