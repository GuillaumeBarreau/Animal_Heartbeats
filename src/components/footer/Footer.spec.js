import { shallow } from 'enzyme';
import React from 'react';
import Footer from './Footer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

const mockStore = configureStore([]);

describe('Footer component', () => {
	let store;

	beforeEach(() => {
		store = mockStore({
			theme: 'dark',
		});
	});

	it('Render Footer component', () => {
		const cmp = shallow(
			<Provider store={store}>
				<Footer />
			</Provider>
		);
		expect(cmp.exists()).toBe(true);
	});

	it('Snapshot testing', () => {
		const cmp = shallow(
			<Provider store={store}>
				<Footer />
			</Provider>
		);
		expect(cmp.html()).toMatchSnapshot();
	});
});
