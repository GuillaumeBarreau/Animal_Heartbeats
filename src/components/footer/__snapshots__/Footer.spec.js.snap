// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component Snapshot testing 1`] = `"<footer class=\\"sc-AykKC llPHYB\\"><p>Made with love ♥ ♥ ♥ - Repository <a target=\\"_blank\\" href=\\"https://github.com/GuillaumeBarreau/Animal_Heartbeats\\" class=\\"sc-AykKD fukFcY\\">GitHub</a></p></footer>"`;
